src/foundations/shape/Shape.mixins.ts

```diff
@@ -0,0 +1,90 @@
+import { css } from 'styled-components';
+import { FlattenSimpleInterpolation } from 'styled-components/ts3.6';
+
+import { Utils } from '../../shared';
+
+import { SHAPE_ELEVATION_DEFINITIONS } from './Shape.constants';
+import { PApplyElevation, PApplyShape } from './Shape.props';
+import { TShapeVariant } from './Shape.types';
+
+/**
+ * apply a shape to a given element without the use of an additional wrapping Shape component.
+ *
+ * @param {(number|string)?} width
+ * @param {(number|string)?} height
+ * @param {TShapeBorderRadius} radius
+ * @returns {FlattenSimpleInterpolation}
+ */
+function applyShape({ width, height, radius }: PApplyShape): FlattenSimpleInterpolation {
+    // define the variant by checking for string
+    const variant: TShapeVariant = Utils.isString(radius) ? radius : 'rectangle';
```

Comment:
what if we pass an invalid string here ie. "random" ?

Reply:
I will add a function to it that asserts correct usage.